metrics-api/src/main/java/software/amazon/swage/metrics/record/MetricRecorder.java

```diff
@@ -90,14 +90,27 @@
      */
     public static class RecorderContext {
         private final TypedMap attributes;
+        private final RecorderContext parent;
 
         /**
          * Constructor mandating the values needed for the emitter-specific public interface.
          *
          * @param attributes the user-supplied attributes of the measurement context
          */
         public RecorderContext(TypedMap attributes) {
+            this(attributes, null);
+        }
+
+         /**
+         *  Constructor for creating a RecorderContext from a parent.
+         *
+         * @param attributes the user-supplied attributes of the measurement context which differentate it 
+         *        from the parent.
+         * @param parent the recorder context parent.
+         */
+        protected RecorderContext(TypedMap attributes, RecorderContext parent) {
```

Comment:
Why not private?

But see other comments - I don't think this is necessarily required at all.